app/controllers/sessions_controller.rb

```diff
@@ -69,24 +69,39 @@ def destroy
     end
   end
 
-  # GET /sessions/verify
-  def verify
-    respond_to do |format|
-      permission = true
-      unless params[:project_id].nil? || params[:verify_owner].nil?
-        owner_id = Project.find(params[:project_id]).user_id
-        verify = params[:verify_owner] == 'true'
-        session_destroyed = session[:user_id].nil?
-        is_admin = (!session_destroyed and User.find(session[:user_id]).admin)
-        permission = (verify && owner_id == session[:user_id]) || !verify || is_admin
+  # GET /sessions/permissions
+  def permissions
+    @permiss = []
+    # Session is null
+    if session[:user_id].nil?
+      respond_to do |format|
+        format.js {}
+        format.json { head :unauthorized }
       end
+      return
+    end
 
-      if session[:user_id].nil? || !permission
-        format.json { render json: '{}', status: :unauthorized }
-      else
-        format.json { render json: '{}', status: :ok }
+    is_admin = User.find(session[:user_id]).admin
+    @permiss.push 'save'
+
+    # Not admin, and you don't own or didn't specify a project
+    unless is_admin
```

Comment:
You basically have an if/else scenario: are you admin or not.  You're doing if not admin, do this and return, otherwise the code continues to the admin condition.  Rather than using negative logic (`unless is_admin`), shouldn't you switch this around so it checks `if is_admin`, do admin stuff, return, otherwise continue on to the non-admin section?